spark/v3.2/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/analysis/RewriteDeleteFromTable.scala

```diff
@@ -49,25 +51,42 @@ import org.apache.spark.sql.execution.datasources.v2.DataSourceV2Relation
 object RewriteDeleteFromTable extends RewriteRowLevelCommand {
 
   override def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators {
-    case d @ DeleteFromIcebergTable(aliasedTable, Some(cond), None) if d.resolved =>
+    case d @ DeleteFromIcebergTable(aliasedTable, Some(_), None) if d.resolved =>
       EliminateSubqueryAliases(aliasedTable) match {
         case r @ DataSourceV2Relation(tbl: SupportsRowLevelOperations, _, _, _, _) =>
-          val operation = buildRowLevelOperation(tbl, DELETE)
-          val table = RowLevelOperationTable(tbl, operation)
-          val rewritePlan = operation match {
-            case _: SupportsDelta =>
-              buildWriteDeltaPlan(r, table, cond)
-            case _ =>
-              buildReplaceDataPlan(r, table, cond)
+          rewriteDeleteFromTable(d, r, tbl)
+        case v: View =>
+          val viewChild = v.child
+          val icebergTableView = v.children.size == 1 && viewChild.isInstanceOf[DataSourceV2Relation] &&
+            viewChild.asInstanceOf[DataSourceV2Relation].table.isInstanceOf[SparkTable]
```

Comment:
Couldn't this be done in a match expression?

I think it would make sense to do this:

```scala
  // omit if EliminateSubqueryAliases works on the whole tree, not just the current node
  EliminateSubqueryAliases(v.child) match {
    case rel @ DataSourceV2Relation(table, Some(_), None) =>
      rewriteDeleteFromTable(d, rel, table)
  }
```

Reply:
Or maybe combine this with the outer case and just have a different way of getting the relation, wrapped by a view.

Reply:
updated